gnucash/gnome/gnc-split-reg.c

```diff
@@ -1054,33 +1055,50 @@ gsr_default_associate_handler_file (GNCSplitReg *gsr, Transaction *trans, gboole
 
     gtk_file_chooser_set_local_only (GTK_FILE_CHOOSER(dialog), FALSE);
 
-    if ((path_head != NULL) && (g_strcmp0 (path_head, "") != 0)) // not default entry
-        valid_path_head = TRUE;
+    if (path_head && g_strcmp0 (path_head, "") != 0) // not default entry
```

Comment:
While you're optimizing the if statement it's probably even more efficient to test for a non-empty string using
```
if (path_head && *path_head != '\0')
```
The dereferencing is safe here as you first verify path_head is a valid non-null pointer.

And going yet one step further you could even eliminate the if statement completely and assign the result of that test directly to path_head_set like so:
```
path_head_set =  (path_head && *path_head != '\0');
```
Of course this is mostly a matter of taste so feel free to take or ignore this suggestion.